src/core/server/licensing/licensing_service.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { BehaviorSubject, Observable, Subscription, from, interval } from 'rxjs';
+import { first, map, switchMap, tap } from 'rxjs/operators';
+import moment from 'moment';
+import { CoreService } from '../../types';
+import { Logger } from '../logging';
+import { CoreContext } from '../core_context';
+import { LicensingServiceSubject, LicensingConfigType, LicensingSetupDependencies } from './types';
+import { SERVICE_NAME, LICENSE_TYPE } from './constants';
+import { LicensingConfig } from './licensing_config';
+import { LicensingServiceSetup } from './licensing_service_setup';
+
+/** @internal */
+export class LicensingService
+  implements CoreService<LicensingServiceSubject, LicensingServiceSubject> {
+  private readonly config$: Observable<LicensingConfig>;
+  private readonly logger: Logger;
+  private poller$!: Observable<number>;
+  private pollerSubscription!: Subscription;
+  private service$!: LicensingServiceSubject;
+  private license: any;
+
+  constructor(private readonly coreContext: CoreContext) {
+    this.logger = coreContext.logger.get(SERVICE_NAME);
+    this.config$ = coreContext.configService
+      .atPath<LicensingConfigType>(SERVICE_NAME)
+      .pipe(map(rawConfig => new LicensingConfig(rawConfig, coreContext.env)));
+  }
+
+  private hasLicenseInfoChanged(newLicense: any) {
+    return (
+      newLicense.mode !== this.license.mode ||
```

Comment:
What's `mode`? If this corresponds to the `LICENSE_TYPE` enum, then maybe `type` would make that clear?

Reply:
This was preexisting from x-pack legacy, so I'll see about making this clearer.